src/model/Spectrum.cpp

```diff
@@ -361,15 +366,43 @@ bool Spectrum::parseMstoolkitSpectrum
   last_scan_ = mst_real_spectrum->getScanNumber();
   precursor_mz_ = mst_real_spectrum->getMZ();
 
+  retention_time_ = mst_real_spectrum->getRTime();
+
   // setfilename of empty spectrum
   filename_ = filename;
 
-  //add all peaks.
-  for(int peak_idx = 0; peak_idx < (int)mst_real_spectrum->size(); peak_idx++){
-    this->addPeak(mst_real_spectrum->at(peak_idx).intensity,
-                   mst_real_spectrum->at(peak_idx).mz);
-  }
-  
+  //add peaks. // This part probably should not run for MS1 data. This should run only for tide-search.
+  // if (Params::GetBool("skip-preprocessing")) {
+    for(int peak_idx = 0; peak_idx < (int)mst_real_spectrum->size(); peak_idx++){
+      this->addPeak(mst_real_spectrum->at(peak_idx).intensity,
+                    mst_real_spectrum->at(peak_idx).mz);
+    }
+    has_peaks_ = true;
+
+  // } else {
```

Comment:
Are these commented lines necessary to keep?

Reply:
done

Reply:
It's an ongoing coding. But I think we will remove it.